pyface/application.py

```diff
@@ -0,0 +1,259 @@
+        # started.  A listener for this event is a good place to do things like
+        # opening application windows and other activities where you want the
+        # event loop running.
+        self.gui.set_trait_later(
+            self, 'application_initialized',
+            ApplicationEvent(application=self, event_type='initialized'))
+
+        # start the GUI - script blocks here
+        self.gui.start_event_loop()
+
+    # Exception handling ------------------------------------------------------
+
+    def _unhandled_exception(self, type, value, traceback):
+        """ Show an error dialog with exception information, if possible. """
+        from pyface.message_dialog import error
+
+        informative = """An unexpected error occurred: {}""".format(
+            format_exception_only(type, value))
+        detail = format_exception(type, value, traceback)
+        error("Unhandled Exception", informative=informative, detail=detail)
+
+    def _excepthook(self, type, value, traceback):
+        """ Handle any exceptions not explicitly caught
+
+        Try to display an appropriate error dialog, if possible.  Fail over to
+        the standard excepthook if that doesn't work.
+        """
+        try:
+            # try to log the exception, could fail eg. if disk full
+            logger.exception(value)
+            self._unhandled_exception(type, value, traceback)
+        except Exception:
+            # something went wrong, use base class excepthook
+            super(Application, self)._excepthook(type, value, traceback)
+        finally:
+            sys.exit(1)
+
+    # Destruction methods -----------------------------------------------------
+
+    def _can_exit(self):
+        """ Check with each window to see if it can be closed """
+        for window in reversed(self.windows):
+            window.closing = event = Vetoable()
+            if event.veto:
+                return False
+        else:
+            return True
+
+    def _prepare_exit(self):
+        """ Close each window """
+        for window in reversed(self.windows):
+            window.destroy()
+            window.closed = True
+
+    def _exit(self):
+        """ Shut down the event loop """
+        self.gui.stop_event_loop()
+
+    # -------------------------------------------------------------------------
+    # Private interface
+    # -------------------------------------------------------------------------
+
+    # Initialization utilities ------------------------------------------------
+
+    def _setup_logging(self):
+        """ Initialize logger.
+
+        By default send logging to a stream and set the log level to info.
+        Most applications will want to replace this with something more
+        appropriate to their needs.
+        """
+        logger = logging.getLogger()
+        logger.addHandler(logging.StreamHandler())
+        logger.setLevel(logging.INFO)
```

Comment:
Personally, I would probably leave this empty, and leave this to the user to set.